docs/msi_v2/guidance_for_sdks_consuming_msal.md

```diff
@@ -0,0 +1,62 @@
+# Guidance for SDKs Consuming MSAL
+
+## Overview
+
+To support MSI V2 authentication with the `/credential` endpoint, the **Azure SDK** leverages the `IMsalMtlsHttpClientFactory` interface and **certificate management APIs** for secure communication with Azure AD using **mutual TLS (mTLS)**.
+
+This section covers:
+- How Azure SDK uses **`IMsalMtlsHttpClientFactory`** for MTLS authentication.
+- How SDKs interact with the **certificate APIs** to obtain the binding certificate certificates.
+- The **new `CertificateRefreshed` event**, which notifies when a binding certificate is updated.
+
+---
+
+## **Handling Certificate Rotation for Long-Lived Clients**
+
+The Problem: Certificate Expiry and Rotation
+
+- mTLS Proof of Possession (PoP) tokens are signed by a binding certificate.
+- The binding certificate is valid for 90 days.
+- A new certificate is made available 5 days before expiration.
+- The SDKs consuming MSAL (customizing httpclient) must ensure that its HttpClient uses the latest certificate.
+
+Proposed Solution
+- SDK clients must monitor certificate updates and refresh their HttpClient dynamically.
+- MSAL exposes an event-driven model to notify when the binding certificate is refreshed.
+
+## **New Interface: `IMsalMtlsHttpClientFactory`**
+
+MSAL introduces the `IMsalMtlsHttpClientFactory` interface to facilitate **mTLS-based authentication** in Azure SDKs. This ensures:
+- Secure token acquisition by enabling **mTLS authentication**.
+- **Reusable `HttpClient` instances** to prevent socket exhaustion.
+- Optimized **Azure SDK integration** for managing identities securely.
+
+### **Interface Definition**
+
+```csharp
+public interface IMsalMtlsHttpClientFactory : IMsalHttpClientFactory
+{
+    /// <summary>
+    /// Returns an HttpClient configured with a certificate for mutual TLS authentication.
+    /// </summary>
+    /// <param name="x509Certificate2">The certificate to be used for MTLS authentication.</param>
+    /// <returns>An HttpClient instance configured with the specified certificate.</returns>
+    HttpClient GetHttpClient(X509Certificate2 x509Certificate2);
```

Comment:
When will this be called? Once per certificate change, or once per auth flow execution?